### PR TITLE
[PBI-1847a] Add unit tests for cacheBuster player param

### PR DESCRIPTION
**Pull request by dchauOoyala**

### CODE REVIEW

File: test/unit-tests/ssai_pulse_test.js [hunk 1]
```diff
@@ -469,4 +469,53 @@ describe('ad_manager_ssai_pulse', function()
...
+    adManagerMetadata =
+    {
+      "cacheBuster": 0
+    };
+    SsaiPulse.loadMetadata(adManagerMetadata, backlotBaseMetadata, movieMetadata);
+    expect(SsaiPulse.bustTheCache).to.be(true);
+
+    adManagerMetadata =
+    {
+      "cacheBuster": false // will not accept boolean
```
Comment (aeng7) [inline test/unit-tests/ssai_pulse_test.js]: We should accept boolean to match the other page level parameters

File: test/unit-tests/ssai_pulse_test.js [hunk 2]
```diff
@@ -469,4 +469,61 @@ describe('ad_manager_ssai_pulse', function()
...
+    };
+    SsaiPulse.loadMetadata(adManagerMetadata, backlotBaseMetadata, movieMetadata);
+    expect(SsaiPulse.bustTheCache).to.be(true);
+
+    adManagerMetadata =
+    {
+      "cacheBuster": "abcd"
+    };
+    SsaiPulse.loadMetadata(adManagerMetadata, backlotBaseMetadata, movieMetadata);
+    expect(SsaiPulse.bustTheCache).to.be(true);
```
Comment (pilievOoyala) [inline test/unit-tests/ssai_pulse_test.js]: is this valid?  Shouldn't you be checking for false?

File: test/unit-tests/ssai_pulse_test.js [hunk 3]
```diff
@@ -469,4 +469,61 @@ describe('ad_manager_ssai_pulse', function()
...
+
+    // should change to false when value is explicitly "false"
+    adManagerMetadata =
+    {
+      "cacheBuster": "false"
+    };
+    SsaiPulse.loadMetadata(adManagerMetadata, backlotBaseMetadata, movieMetadata);
+    expect(SsaiPulse.bustTheCache).to.be(false);
+  });
```
Comment (pilievOoyala) [inline test/unit-tests/ssai_pulse_test.js]: test setting to undefined null and an empty object

File: js/ssai_pulse.js [hunk 4]
```diff
@@ -144,22 +144,33 @@ OO.Ads.manager(function(_, $)
 
       if (adManagerMetadata)
       {
-        if (adManagerMetadata["cacheBuster"])
+        // allow boolean true/false
+        if (_.isBoolean(adManagerMetadata["cacheBuster"]))
         {
-          if (adManagerMetadata["cacheBuster"] === "true")
+          if (adManagerMetadata["cacheBuster"])
```
Comment (aeng7) [inline js/ssai_pulse.js]: If we've determined that adManagerMetadata["cacheBuster"] is a boolean, can we set this.bustTheCache to this?
this.bustTheCache = adManagerMetadata["cacheBuster"];

**Pull request merged**